### PR TITLE
Consistently reset color formatting

### PR DESCRIPTION
**Pull request by thaliaarchi**
Before, arrays would not reset colors after `[` and `,`, but objects would; it would reset colors twice before `]` and `}`; and some cases of indentation would have colors applied. Now, colors are reset immediately after any token that is colored, before any indentation. This makes the formatting consistent, for the benefit of custom `JQ_COLORS`.

### CODE REVIEW

File: tests/shtest
Comment (wader) [inline tests/shtest:508]: Does this regression test verify the reset issue for arrays? i can visually see the issue using `JQ_COLORS='0;30:0;31:0;32:0;33:0;34:7;35:1;36:1;37' jq -Cn '[{"a":true,"b":false},123,null]'` (7 to reverse colors for arrays)

Nice tests 👍

**Review by wader — APPROVED** [review]
LGTM but let's wait for one more review

File: tests/shtest
Comment (emanuele6) [reply tests/shtest:508]: file

**Review by emanuele6 — APPROVED** [review]
LGTM. Thank you!

**Comment by thaliaarchi** [comment]
The only remaining inconsistency, is that `: ` is one unit for coloring. So that space would be the only one with a background color. I could make it clear before the colon. I don’t have a strong preference either way, but now I’m leaning towards clear between colon and space. Do we want the background to fill the space between the key and value or just be behind the colon?

I’ll push that fix later in the day.

**Comment by emanuele6** [comment]
For future reference
file

**Comment by thaliaarchi** [comment]
Thanks for the screenshots. They've inspired me to make my own, and here's the color formatting over the relevant commits:

```sh
JQ_COLORS='48;5;89:48;5;21:48;5;28:48;5;207:48;5;51:48;5;9:48;5;208:48;5;11' \
jq -n '[{"a":true,"b":false},"abc",123,null]'
```

color-reset

**Comment by thaliaarchi** [comment]
I'm all done on my end and am ready to merge.

Here's a before/after recap:

```diff
< ␛[1;35m[\n
> ␛[1;35m[␛[0m\n
<   ␛[1;36m{\n
>   ␛[1;36m{␛[0m\n
<     ␛[0m␛[1;37m"a"␛[0m␛[1;36m: ␛[0m␛[0;32mtrue␛[0m␛[1;36m,\n
>     ␛[1;37m"a"␛[0m␛[1;36m:␛[0m ␛[0;32mtrue␛[0m␛[1;36m,␛[0m\n
<     ␛[0m␛[1;37m"b"␛[0m␛[1;36m: ␛[0m␛[0;31mfalse␛[0m␛[1;36m\n
>     ␛[1;37m"b"␛[0m␛[1;36m:␛[0m ␛[0;31mfalse␛[0m\n
<   ␛[1;36m}␛[0m␛[1;35m,\n
>   ␛[1;36m}␛[0m␛[1;35m,␛[0m\n
<   ␛[0;34m"abc"␛[0m␛[1;35m,\n
>   ␛[0;34m"abc"␛[0m␛[1;35m,␛[0m\n
<   ␛[0;33m123␛[0m␛[1;35m,\n
>   ␛[0;33m123␛[0m␛[1;35m,␛[0m\n
<   ␛[0;30mnull␛[0m␛[1;35m\n
>   ␛[0;30mnull␛[0m\n
< ␛[1;35m]␛[0m\n
> ␛[1;35m]␛[0m\n
```

Before:

```
␛[1;35m[\n
  ␛[1;36m{\n
    ␛[0m␛[1;37m"a"␛[0m␛[1;36m: ␛[0m␛[0;32mtrue␛[0m␛[1;36m,\n
    ␛[0m␛[1;37m"b"␛[0m␛[1;36m: ␛[0m␛[0;31mfalse␛[0m␛[1;36m\n
  ␛[1;36m}␛[0m␛[1;35m,\n
  ␛[0;34m"abc"␛[0m␛[1;35m,\n
  ␛[0;33m123␛[0m␛[1;35m,\n
  ␛[0;30mnull␛[0m␛[1;35m\n
␛[1;35m]␛[0m\n
```

After:

```
␛[1;35m[␛[0m\n
  ␛[1;36m{␛[0m\n
    ␛[1;37m"a"␛[0m␛[1;36m:␛[0m ␛[0;32mtrue␛[0m␛[1;36m,␛[0m\n
    ␛[1;37m"b"␛[0m␛[1;36m:␛[0m ␛[0;31mfalse␛[0m\n
  ␛[1;36m}␛[0m␛[1;35m,␛[0m\n
  ␛[0;34m"abc"␛[0m␛[1;35m,␛[0m\n
  ␛[0;33m123␛[0m␛[1;35m,␛[0m\n
  ␛[0;30mnull␛[0m\n
␛[1;35m]␛[0m\n
```

Generated roughly with:

```bash
#!/usr/bin/env bash
set -euo pipefail

for ref in 8a9a74d ccebd8e; do
  git checkout "$ref"
  make -j8
  mv jq "$(git describe --tags)"
done

colors() {
  JQ_COLORS='0;30:0;31:0;32:0;33:0;34:1;35:1;36:1;37' \
  "$1" -Cn '[{"a":true,"b":false},"abc",123,null]' |
    gsed -z -e 's/\n/\\n\n/g' -e 's/\x1b/␛/g'
}
diff <(colors ./jq-1.7.1-18-g8a9a74d) \
     <(colors ./jq-1.7.1-21-gccebd8e)
```

**Review by wader — APPROVED** [review]
👍

**Comment by nicowilliams** [comment]
Thanks!

**Pull request merged**
